packages/multichain-network-controller/src/api/accounts-api.test.ts

```diff
@@ -0,0 +1,265 @@
+import {
+  BtcScope,
+  SolScope,
+  EthScope,
+  EthAccountType,
+  BtcAccountType,
+  SolAccountType,
+} from '@metamask/keyring-api';
+import {
+  type Json,
+  type CaipAccountId,
+  KnownCaipNamespace,
+} from '@metamask/utils';
+
+import {
+  type ActiveNetworksResponse,
+  toAllowedCaipAccountIds,
+  toActiveNetworksByAddress,
+  buildActiveNetworksUrl,
+  MULTICHAIN_ACCOUNTS_BASE_URL,
+} from './accounts-api';
+
+const MOCK_ADDRESSES: {
+  evm: string;
+  solana: string;
+  bitcoin: string;
+} = {
+  evm: '0x1234567890123456789012345678901234567890',
+  solana: 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v',
+  bitcoin: 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq',
+};
+
+const MOCK_CAIP_IDS: {
+  evm: CaipAccountId;
+  solana: CaipAccountId;
+  bitcoin: CaipAccountId;
+  invalid: string;
+  invalidEvm: string;
+  invalidSolana: string;
+  invalidBitcoin: string;
+  unsupportedNamespace: string;
+  customNamespace: string;
+  testNamespace: string;
+} = {
+  evm: `${KnownCaipNamespace.Eip155}:1:${MOCK_ADDRESSES.evm}`,
+  solana: `${KnownCaipNamespace.Solana}:1:${MOCK_ADDRESSES.solana}`,
+  bitcoin: `${KnownCaipNamespace.Bip122}:1:${MOCK_ADDRESSES.bitcoin}`,
+  invalid: 'invalid:format:address',
+  invalidEvm: `${KnownCaipNamespace.Eip155}:1:0xinvalid`,
+  invalidSolana: `${KnownCaipNamespace.Solana}:1:invalid`,
+  invalidBitcoin: `${KnownCaipNamespace.Bip122}:1:invalid`,
+  unsupportedNamespace:
+    'cosmos:1:cosmos1hsk6jryyqjfhp5dhc55tc9jtckygx0eph6dd02',
+  customNamespace: `custom:1:${MOCK_ADDRESSES.evm}`,
+  testNamespace: `test:1:${MOCK_ADDRESSES.evm}`,
+};
+
+describe('toAllowedCaipAccountIds', () => {
+  type AccountType = {
+    type: EthAccountType | BtcAccountType | SolAccountType;
+    id: string;
+    options: Record<string, Json>;
+    methods: string[];
+    metadata: {
+      name: string;
+      importTime: number;
+      keyring: { type: string };
+    };
+    address: string;
+    scopes: `${string}:${string}`[];
+  };
+
+  const createMockAccount = (
+    address: string,
+    scopes: `${string}:${string}`[],
+    type: EthAccountType | BtcAccountType | SolAccountType,
+  ): AccountType => ({
+    address,
+    scopes,
+    type,
+    id: '1',
+    options: {},
+    methods: [],
+    metadata: {
+      name: 'Test Account',
+      importTime: Date.now(),
+      keyring: { type: 'test' },
+    },
+  });
+
+  it('formats account with EVM scopes', () => {
+    const account = createMockAccount(
+      MOCK_ADDRESSES.evm,
+      [EthScope.Eoa, EthScope.Mainnet, EthScope.Testnet],
+      EthAccountType.Eoa,
+    );
+
+    const result = toAllowedCaipAccountIds(account);
+    expect(result).toStrictEqual([
+      `${EthScope.Eoa}:${MOCK_ADDRESSES.evm}`,
```

Comment:
I would probably remove the `Eoa` case here. The main reason being that the API would never return and address with the EOA scope, since we want to know the activity on a given chain and `eip155:0` means every EVM chains 😅

And I wonder if we should prevent using EOA in our logic 🤔 for now, we should trust the API and do not have any conditional logic, I'd still remove from the test to avoid any confusion:

 ```suggestion
      [EthScope.Mainnet, EthScope.Testnet],
      EthAccountType.Eoa,
    );

    const result = toAllowedCaipAccountIds(account);
    expect(result).toStrictEqual([
```